### PR TITLE
Add Kearas2 GlobalMaxPooling1D   Layer and Test

### PR DESCRIPTION
**Pull request by yjjiang11**

### CODE REVIEW

File: pyzoo/zoo/pipeline/api/keras2/layers/pooling.py [hunk 1]
```diff
@@ -21,7 +21,40 @@
...
+        super(AveragePooling1D, self).__init__(None,
+                                               pool_length,
+                                               stride,
+                                               border_mode,
+                                               list(input_shape) if input_shape else None,
+                                               **kwargs)
+
+
 class MaxPooling1D(ZooKeras2Layer):
+
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras2/layers/pooling.py]: remove this line?

File: pyzoo/zoo/pipeline/api/keras2/layers/pooling.py [hunk 2]
```diff
@@ -21,7 +21,40 @@
...
+    The input of this layer should be 3D.
+
+    When you use this layer as the first layer of a model, you need to provide the argument
+    input_shape (a shape tuple, does not include the batch dimension).
+
+    # Arguments
+    pool_length: Size of the region to which max pooling is applied.
+    strides: Factor by which to downscale. 2 will halve the input.
+             Default is None, and in this case it will be equal to pool_length..
+    border_mode: Either 'valid' or 'same'. Default is 'valid'.
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras2/layers/pooling.py]: the name should be padding?

File: pyzoo/zoo/pipeline/api/keras2/layers/pooling.py [hunk 3]
```diff
@@ -21,7 +21,40 @@
     unicode = str
 
 
+class AveragePooling1D(ZooKeras2Layer):
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras2/layers/pooling.py]: Also update doc for MaxPooling1D?

File: pyzoo/zoo/pipeline/api/keras2/layers/pooling.py [hunk 4]
```diff
@@ -21,7 +21,40 @@
...
+             Default is None, and in this case it will be equal to pool_length..
+    border_mode: Either 'valid' or 'same'. Default is 'valid'.
+    input_shape: A shape tuple, not including batch.
+    name: String to set the name of the layer.
+          If not specified, its name will by default to be a generated string.
+
+    >>> averagepooling1d = AveragePooling1D(input_shape=(3, 24))
+    creating: createZooKeras2AveragePooling1D
+    """
+    def __init__(self, pool_length=2, stride=None, border_mode="valid",
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras2/layers/pooling.py]: padding? Check keras2 docs.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalMaxPooling1D.scala [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  * Applies global max pooling operation for temporal data.
+  * The input of this layer should be 3D.
+  *
+  * When you use this layer as the first layer of a model, you need to provide the argument
+  * inputShape (a Single Shape, does not include the batch dimension).
+  *
+  * @param inputShape A Single Shape, does not include the batch dimension.
+  * @tparam T The numeric type of parameter(e.g. weight, bias). Only support float/double now.
+  */
+
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalMaxPooling1D.scala]: remove this line?

**Comment by yjjiang11** [comment]
Jenkins passed : http://172.168.2.101:8080/job/ZOO-PR-Python-Spark-1.6-py27/730/ @hkvision

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/python/PythonZooKeras2.scala [hunk 6]
```diff
@@ -22,12 +22,12 @@ import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
...
 import com.intel.analytics.bigdl.python.api.PythonBigDLKeras
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
-import com.intel.analytics.zoo.pipeline.api.Net
-import com.intel.analytics.zoo.pipeline.api.autograd.Variable
-import com.intel.analytics.zoo.pipeline.api.keras2.layers.{Conv1D, Dense, MaxPooling1D, AveragePooling1D, Maximum, Minimum}
-import scala.collection.JavaConverters._
 
 
+import com.intel.analytics.zoo.pipeline.api.keras2.layers.{Conv1D, Dense, MaxPooling1D, AveragePooling1D, Maximum, Minimum}
+_
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/python/PythonZooKeras2.scala]: what's this?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/python/PythonZooKeras2.scala [hunk 7]
```diff
@@ -22,12 +22,12 @@ import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
 import com.intel.analytics.bigdl.optim.Regularizer
 import com.intel.analytics.bigdl.python.api.PythonBigDLKeras
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
-import com.intel.analytics.zoo.pipeline.api.Net
-import com.intel.analytics.zoo.pipeline.api.autograd.Variable
-import com.intel.analytics.zoo.pipeline.api.keras2.layers.{Conv1D, Dense, MaxPooling1D, AveragePooling1D, Maximum, Minimum}
-import scala.collection.JavaConverters._
 
 
+import com.intel.analytics.zoo.pipeline.api.keras2.layers.{Conv1D, Dense, MaxPooling1D, AveragePooling1D, Maximum, Minimum}
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/python/PythonZooKeras2.scala]: as you have already imported everything in Line29, this line can be deleted.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/python/PythonZooKeras2.scala [hunk 8]
```diff
@@ -121,5 +121,17 @@ class PythonZooKeras2[T: ClassTag](implicit ev: TensorNumeric[T]) extends Python
       toScalaShape(inputShape))
   }
 
+  def createZooKeras2GlobalAveragePooling1D(
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/python/PythonZooKeras2.scala]: Haven't you added this in the PR for GlobelAveragePooling1D?

**Pull request closed without merging**